archive/o/odin/longest-word.odin

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "core:fmt"
+import "core:strings"
+import "core:os"
+
+main :: proc() {
+    if len(os.args) != 2 {
+        usage()
+        return
+    }
+    text := os.args[1]
+    get_longest_word(text)
+}
+
+get_longest_word :: proc (word: string) -> int {
+    if len(strings.trim(word, "\t\r\n ")) == 0 {
+        usage()
+        return 0
+    }
+    string_split := strings.split(word, " ", context.allocator)
```

Comment:
I think I see the problem. With the `"Hi,\nMy name is Paul!"` case, the first word is not separated by a space character. Since this language doesn't seem to have a way to split on multiple characters (at least I couldn't find one) and doesn't seem to support regular expressions, I'd recommend converting each whitespace character to a space, then split on space.